api/src/main/java/org/apache/iceberg/expressions/Literals.java

```diff
@@ -57,6 +58,7 @@ private Literals() {
   @SuppressWarnings("unchecked")
   static <T> Literal<T> from(T value) {
     Preconditions.checkNotNull(value, "Cannot create expression literal from null");
+    Preconditions.checkArgument(!NaNUtil.isNaN(value), "Cannot expression literal from NaN");
```

Comment:
Looks like the verb is missing. It should be "Cannot create expression literal from NaN".

Reply:
Yes, sorry should have double checked... I include the fix of this in #1857 since I think they are kind of related, but please let me know if you want a separate PR for that!